girder_worker/plugins/docker/executor.py

```diff
@@ -118,6 +118,26 @@ def _docker_gc(tempdir):
     return subprocess.Popen(args=(script,), env=env)
 
 
+def validate_task_outputs(task_outputs):
+    """
+    This is called prior to fetching inputs to make sure the output specs are
+    valid. Outputs in docker mode can result in side effects, so it's best to
+    make sure the specs are valid prior to fetching.
+    """
+    for name, spec in task_outputs.iteritems():
+        if spec.get('target') == 'filepath':
+            if 'path' not in spec:
```

Comment:
Also if you do want them to specify `path`, maybe you should ask them to specify it in the binding spec? I feel this is kind of a binding (where will the data be produced) specification rather than the task specification.